### PR TITLE
Update kafka.go to add transport.kafka.compression.type as an option

### PR DESCRIPTION
**Pull request by acherla**
Addresses Issue: https://github.com/netsampler/goflow2/issues/92

### CODE REVIEW

File: transport/kafka/kafka.go [hunk 1]
```diff
@@ -54,6 +55,7 @@ func (d *KafkaDriver) Prepare() error {
 
 	//flag.StringVar(&d.kafkaKeying, "transport.kafka.key", "SamplerAddress,DstAS", "Kafka list of fields to do hashing on (partition) separated by commas")
 	flag.StringVar(&d.kafkaVersion, "transport.kafka.version", "2.8.0", "Kafka version")
+	flag.StringVar(&d.kafkaCompressionType, "transport.kafka.compression.type", "lz4", "Kafka default compression type")
```
Comment (lspgn) [inline transport/kafka/kafka.go:58]: Could you leave it empty, seems that we are introducing a default.

File: transport/kafka/kafka.go [hunk 2]
```diff
@@ -71,6 +73,8 @@ func (d *KafkaDriver) Init(context.Context) error {
 	kafkaConfig.Producer.MaxMessageBytes = d.kafkaMaxMsgBytes
 	kafkaConfig.Producer.Flush.Bytes = d.kafkaFlushBytes
 	kafkaConfig.Producer.Flush.Frequency = d.kafkaFlushFrequency
+	kafkaConfig.Producer.Compression = sarama.CompressionCodec(d.kafkaCompressionType)
```
Comment (lspgn) [inline transport/kafka/kafka.go:76]: Does it handle the case where it's empty?

Comment (lspgn) [reply transport/kafka/kafka.go:76]: And what is the behavior if user input is not recognized?
Should we raise an error?

Comment (acherla) [reply transport/kafka/kafka.go:76]: We can probably let samsa raise the error if its not recognized

File: transport/kafka/kafka.go (hunk 1 above)
Comment (acherla) [reply transport/kafka/kafka.go:58]: we can set the type=none by default to not enforce a new default parameter to the goflow2 producer

File: transport/kafka/kafka.go (hunk 2 above)
Comment (acherla) [reply transport/kafka/kafka.go:76]: Both situations IMO should be handled by samsa, we should merely provide a default value of "none" if no parameter is provided

File: transport/kafka/kafka.go (hunk 1 above)
Comment (acherla) [reply transport/kafka/kafka.go]: change made

**Comment by acherla** [comment]
Fixed build issue @lspgn can you re-run workflow again?

File: README.md [hunk 3]
```diff
@@ -150,6 +150,25 @@ To partition the feed (any field of the protobuf is available), the following op
...
+transport.kafka.compression.type=3
+```
+Optionally to disable compression on the producer end configure the below parameter
+```
+transport.kafka.compression.type=0
+```
+
+List of all compression types:
+```
+CompressionNone   CompressionCodec = 0
```
Comment (lspgn) [inline README.md:165]: Hmm we should keep the lookup. Just need to handle the case where the mapping is not found.

I think we could use [UnmarshalText](https://pkg.go.dev/github.com/Shopify/sarama#CompressionCodec.UnmarshalText)

**Comment by lspgn** [comment]
Implemented the changes as part of #128 (not sure if there's a better way to do with GitHub). Closing this PR.

Thank you!

**Pull request closed without merging**